### PR TITLE
New Adapter: TeqBlaze

### PR DESCRIPTION
**Pull request by MaksymTeqBlaze**

### CODE REVIEW

File: static/bidder-info/teqblaze.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+whiteLabelOnly: true
+maintainer:
+  email: "github@teqblaze.com"
```
Comment (bsardo) [inline static/bidder-info/teqblaze.yaml:3]: We sent an email to this address. Please respond with a message of "received" to confirm the address is reachable.

File: adapters/teqblaze/teqblaze.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	reqJSON, err := jsonutil.Marshal(request)
+	if err != nil {
+		return nil, err
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	return &adapters.RequestData{
+		Method:  "POST",
```
Comment (bsardo) [inline adapters/teqblaze/teqblaze.go]: Nitpick: use `http.MethodPost`

File: adapters/teqblaze/teqblaze.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+	for _, seatBid := range response.SeatBid {
+		for i := range seatBid.Bid {
+			bid := seatBid.Bid[i]
+			bidType, err := getBidType(bid)
+			if err != nil {
+				return nil, []error{err}
+			}
+
+			b := &adapters.TypedBid{
+				Bid:     &seatBid.Bid[i],
```
Comment (bsardo) [inline adapters/teqblaze/teqblaze.go]: Nitpick: you can just make this `&bid`. Alternatively you can delete the line `bid := seatBid.Bid[i]` above and keep what you have here.

File: adapters/teqblaze/params_test.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	`{"placementId": "test"}`,
+	`{"placementId": "1"}`,
+	`{"endpointId": "test"}`,
+	`{"endpointId": "1"}`,
+}
+
+var invalidParams = []string{
+	`{"placementId": 42}`,
+	`{"endpointId": 42}`,
+	`{"placementId": "1", "endpointId": "1"}`,
```
Comment (bsardo) [inline adapters/teqblaze/params_test.go:46]: Please also add the following to test the minimum length checks:
```
`{"placementId": ""}`,
`{"endpointId": ""}`,
```

File: adapters/teqblaze/teqblaze.go [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+	reqCopy := *request
+	for _, imp := range request.Imp {
+		reqCopy.Imp = []openrtb2.Imp{imp}
+
+		var bidderExt adapters.ExtImpBidder
+		var teqblazeExt openrtb_ext.ImpExtTeqBlaze
+
+		if err := jsonutil.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errs = append(errs, err)
```
Comment (bsardo) [inline adapters/teqblaze/teqblaze.go:47]: Please add supplemental JSON test verifying the unmarshal error is properly handled. You can do this by making a malformed `imp.ext` by setting it to a type other than an object.

File: adapters/teqblaze/teqblaze.go [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+			continue
+		}
+		if err := jsonutil.Unmarshal(bidderExt.Bidder, &teqblazeExt); err != nil {
+			errs = append(errs, err)
```
Comment (bsardo) [inline adapters/teqblaze/teqblaze.go:51]: Please add supplemental JSON test verifying the unmarshal error is properly handled. You can do this by making a malformed `imp.ext.bidder` by setting it to a type other than an object:
```
{
    "mockBidRequest": {
        "id": "test-request-id",
        "device": {
            "ip": "123.123.123.123",
            "ua": "iPad"
        },
        "app": {
            "id": "1",
            "bundle": "com.wls.testwlsapplication"
        },
        "imp": [
            {
                "id": "test-imp-id",
                "tagid": "test",
                "banner": {
                    "format": [
                        {
                            "w": 300,
                            "h": 250
                        },
                        {
                            "w": 300,
                            "h": 600
                        }
                    ]
                },
                "ext": {
                    "bidder": 123
                }
            }
        ]
    },
    "expectedMakeRequestsErrors": [
    {
      "value": "expect { or n, but found 1",
      "comparison": "literal"
    }
  ]
}

```

File: adapters/teqblaze/teqblazetest/exemplary/multi_imp.json [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+                    ],
+                    "w": 1024,
+                    "h": 576
+                },
+                "ext": {
+                    "bidder": {
+                        "endpointId": "test"
+                    }
+                }
+            }
```
Comment (bsardo) [inline adapters/teqblaze/teqblazetest/exemplary/multi_imp.json]: The name of this test indicates it should be a multi-imp test but there is only one imp. Can you add another imp to it?

File: adapters/teqblaze/teqblazetest/exemplary/endpoint_id.json [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+                                    "w": 300,
+                                    "h": 250,
+                                    "ext": {
+                                        "prebid": {
+                                            "type": "banner"
+                                        }
+                                    }
+                                }
+                            ],
+                            "seat": "teqlaze"
```
Comment (bsardo) [inline adapters/teqblaze/teqblazetest/exemplary/endpoint_id.json]: Super nitpick: all of your tests are mispelling teqblaze as the seat.

File: static/bidder-info/teqblaze.yaml (hunk 1 above)
Comment (bsardo) [reply static/bidder-info/teqblaze.yaml:3]: Email received. Confirmed.

**Review by bsardo — APPROVED** [review]
